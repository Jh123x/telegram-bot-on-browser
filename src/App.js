import { Fragment, useEffect } from "react";
import { CssBaseline, Container } from "@mui/material";
import { TokenSaver } from "./component/TokenSaver";
import { Navbar } from "./component/Navbar";
import { BotInterface } from "./component/BotInterface";
import { BotOperation } from "./component/BotOperation";
import { setToken, setCommands } from "./redux/botSlice";
import { useDispatch } from "react-redux";

export const App = () => {
  const dispatch = useDispatch();
  useEffect(() => {
    const token = localStorage.getItem("token");
    if (token !== null) dispatch(setToken(token));
    const commands = localStorage.getItem("commands");
    if (commands !== null) dispatch(setCommands(JSON.parse(commands)));
  }, [dispatch]);
  return (
    <Fragment>
      <CssBaseline />
      <Navbar />
      <Container>
        <TokenSaver />
        <BotInterface />
        <BotOperation />
      </Container>
    </Fragment>
  );
};
export default App;
