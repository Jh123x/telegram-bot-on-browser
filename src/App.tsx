import { useEffect } from "react";
import { useDispatch } from "react-redux";
import { CssBaseline, Container } from "@mui/material";

import { TokenSaver } from "./component/TokenSaver.tsx";
import { Navbar } from "./component/Navbar.tsx";
import { BotInterface } from "./component/BotInterface.tsx";
import { BotOperation } from "./component/BotOperation.tsx";
import { setToken, setCommands } from "./redux/botSlice.ts";
import { Footer } from "./component/Footer.tsx";
import React from "react";
import { LogBox } from "./component/logs.tsx";

export const App = () => {
  const dispatch = useDispatch();
  useEffect(() => {
    const token = localStorage.getItem("token");
    if (token !== null) dispatch(setToken(token));
    const commands = localStorage.getItem("commands");
    if (commands !== null) dispatch(setCommands(JSON.parse(commands)));
  }, [dispatch]);
  return (
    <Container sx={{
      height: '100%',
      maxWidth: '100%',
      width: '100%',
      minWidth: '100%',
      margin: 0,
      padding: 0,
    }} disableGutters={true}>
      <CssBaseline />
      <Navbar />
      <Container>
        <TokenSaver />
        <BotInterface />
        <BotOperation />
        <LogBox />
      </Container>
      <Footer />
    </Container>
  );
};
export default App;
