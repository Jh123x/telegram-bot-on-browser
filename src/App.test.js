import {
  generateDefaultState,
  renderWithProviders,
  setupStore,
} from "./redux/testUtils";
import App from "./App";

test("renders app correctly", () => {
  const store = setupStore(generateDefaultState());
  const component = renderWithProviders(<App />, { store: store });
  expect(component).toMatchSnapshot();
});
