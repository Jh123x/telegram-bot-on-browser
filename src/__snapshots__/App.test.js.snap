// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders app correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-ptdwpq-MuiPaper-root-MuiAppBar-root"
        style="--Paper-shadow: 0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12);"
      >
        <div
          class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1ygil4i-MuiToolbar-root"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 css-1miy0lu-MuiTypography-root"
          >
            BrowserBot
          </h6>
        </div>
      </header>
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg css-5c1adp-MuiContainer-root"
      >
        <div
          class="MuiFormGroup-root css-1mac7pt-MuiFormGroup-root"
        >
          <h3
            class="MuiTypography-root MuiTypography-h3 css-13msox4-MuiTypography-root"
          >
            API Token
          </h3>
          <p
            class="MuiTypography-root MuiTypography-body1 css-rizt0-MuiTypography-root"
          >
            Please enter your API token to save it to local storage.
          </p>
          <br />
          <p
            class="MuiTypography-root MuiTypography-body1 css-rizt0-MuiTypography-root"
          >
            API Token
          </p>
          <div
            class="MuiFormControl-root MuiTextField-root css-1xp5r68-MuiFormControl-root-MuiTextField-root"
          >
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-quhxjy-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input css-16wblaj-MuiInputBase-input-MuiOutlinedInput-input"
                id=":r0:"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-w4cd9x"
                >
                  <span
                    aria-hidden="true"
                    class="notranslate"
                  >
                    â€‹
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-74d805-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            Save
          </button>
          <br />
        </div>
        <h3
          class="MuiTypography-root MuiTypography-h3 css-13msox4-MuiTypography-root"
        >
          Commands
        </h3>
        <p
          class="MuiTypography-root MuiTypography-body1 css-rizt0-MuiTypography-root"
        >
          When a user sends \`Command\`, the bot will reply with \`Response\`.
        </p>
        <div
          class="MuiTableContainer-root css-70zvr5-MuiTableContainer-root"
        >
          <table
            class="MuiTable-root css-1ks3uzw-MuiTable-root"
          >
            <thead
              class="MuiTableHead-root css-1a7iywq-MuiTableHead-root"
            >
              <tr
                class="MuiTableRow-root MuiTableRow-head css-16psxo0-MuiTableRow-root"
              >
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeMedium css-to33az-MuiTableCell-root"
                  scope="col"
                >
                  Command
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeMedium css-to33az-MuiTableCell-root"
                  scope="col"
                >
                  Response
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeMedium css-to33az-MuiTableCell-root"
                  scope="col"
                >
                  Delete
                </th>
              </tr>
            </thead>
            <tbody
              class="MuiTableBody-root css-gmh7jj-MuiTableBody-root"
            />
          </table>
          <div
            class="MuiBox-root css-uy5y4h"
          />
        </div>
        <div
          class="MuiFormGroup-root css-1mac7pt-MuiFormGroup-root"
        >
          <div
            class="MuiFormControl-root MuiTextField-root css-1xp5r68-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-19qnlrw-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for=":r2:"
              id=":r2:-label"
            >
              Command
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-quhxjy-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input css-16wblaj-MuiInputBase-input-MuiOutlinedInput-input"
                id=":r2:"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-81qg8w"
                >
                  <span>
                    Command
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <br />
          <div
            class="MuiFormControl-root MuiTextField-root css-1xp5r68-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-19qnlrw-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for=":r3:"
              id=":r3:-label"
            >
              Response
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-multiline css-xrmkj5-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <textarea
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline css-w4nesw-MuiInputBase-input-MuiOutlinedInput-input"
                id=":r3:"
                style="height: 0px; overflow: hidden;"
              />
              <textarea
                aria-hidden="true"
                class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline css-w4nesw-MuiInputBase-input-MuiOutlinedInput-input"
                readonly=""
                style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); padding-top: 0px; padding-bottom: 0px; width: 100%;"
                tabindex="-1"
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-81qg8w"
                >
                  <span>
                    Response
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <br />
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-74d805-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            Add
          </button>
        </div>
        <br />
        <p
          class="MuiTypography-root MuiTypography-body1 css-rizt0-MuiTypography-root"
        >
          Bot is 
          stopped
        </p>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSuccess MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorSuccess MuiButton-root MuiButton-contained MuiButton-containedSuccess MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorSuccess css-1elmdwi-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Start
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedError MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorError Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedError MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorError css-1f7qp0x-MuiButtonBase-root-MuiButton-root"
          disabled=""
          tabindex="-1"
          type="button"
        >
          Stop
        </button>
      </div>
      <footer
        class="footer MuiBox-root css-0"
      >
        <div
          class="MuiContainer-root MuiContainer-maxWidthSm css-zzzaw2-MuiContainer-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body2 text_box css-7ctihp-MuiTypography-root"
          >
            Copyright Â©
            <a
              class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways url css-i9kmoy-MuiTypography-root-MuiLink-root"
              href="https://jh123x.com/"
            >
              Jh123x
            </a>
            2025
          </p>
        </div>
      </footer>
    </div>
  </body>,
  "container": <div>
    <header
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-ptdwpq-MuiPaper-root-MuiAppBar-root"
      style="--Paper-shadow: 0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12);"
    >
      <div
        class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1ygil4i-MuiToolbar-root"
      >
        <h6
          class="MuiTypography-root MuiTypography-h6 css-1miy0lu-MuiTypography-root"
        >
          BrowserBot
        </h6>
      </div>
    </header>
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg css-5c1adp-MuiContainer-root"
    >
      <div
        class="MuiFormGroup-root css-1mac7pt-MuiFormGroup-root"
      >
        <h3
          class="MuiTypography-root MuiTypography-h3 css-13msox4-MuiTypography-root"
        >
          API Token
        </h3>
        <p
          class="MuiTypography-root MuiTypography-body1 css-rizt0-MuiTypography-root"
        >
          Please enter your API token to save it to local storage.
        </p>
        <br />
        <p
          class="MuiTypography-root MuiTypography-body1 css-rizt0-MuiTypography-root"
        >
          API Token
        </p>
        <div
          class="MuiFormControl-root MuiTextField-root css-1xp5r68-MuiFormControl-root-MuiTextField-root"
        >
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-quhxjy-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input css-16wblaj-MuiInputBase-input-MuiOutlinedInput-input"
              id=":r0:"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-w4cd9x"
              >
                <span
                  aria-hidden="true"
                  class="notranslate"
                >
                  â€‹
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-74d805-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Save
        </button>
        <br />
      </div>
      <h3
        class="MuiTypography-root MuiTypography-h3 css-13msox4-MuiTypography-root"
      >
        Commands
      </h3>
      <p
        class="MuiTypography-root MuiTypography-body1 css-rizt0-MuiTypography-root"
      >
        When a user sends \`Command\`, the bot will reply with \`Response\`.
      </p>
      <div
        class="MuiTableContainer-root css-70zvr5-MuiTableContainer-root"
      >
        <table
          class="MuiTable-root css-1ks3uzw-MuiTable-root"
        >
          <thead
            class="MuiTableHead-root css-1a7iywq-MuiTableHead-root"
          >
            <tr
              class="MuiTableRow-root MuiTableRow-head css-16psxo0-MuiTableRow-root"
            >
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeMedium css-to33az-MuiTableCell-root"
                scope="col"
              >
                Command
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeMedium css-to33az-MuiTableCell-root"
                scope="col"
              >
                Response
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeMedium css-to33az-MuiTableCell-root"
                scope="col"
              >
                Delete
              </th>
            </tr>
          </thead>
          <tbody
            class="MuiTableBody-root css-gmh7jj-MuiTableBody-root"
          />
        </table>
        <div
          class="MuiBox-root css-uy5y4h"
        />
      </div>
      <div
        class="MuiFormGroup-root css-1mac7pt-MuiFormGroup-root"
      >
        <div
          class="MuiFormControl-root MuiTextField-root css-1xp5r68-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-19qnlrw-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for=":r2:"
            id=":r2:-label"
          >
            Command
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-quhxjy-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input css-16wblaj-MuiInputBase-input-MuiOutlinedInput-input"
              id=":r2:"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-81qg8w"
              >
                <span>
                  Command
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <br />
        <div
          class="MuiFormControl-root MuiTextField-root css-1xp5r68-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-19qnlrw-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for=":r3:"
            id=":r3:-label"
          >
            Response
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-multiline css-xrmkj5-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <textarea
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline css-w4nesw-MuiInputBase-input-MuiOutlinedInput-input"
              id=":r3:"
              style="height: 0px; overflow: hidden;"
            />
            <textarea
              aria-hidden="true"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline css-w4nesw-MuiInputBase-input-MuiOutlinedInput-input"
              readonly=""
              style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); padding-top: 0px; padding-bottom: 0px; width: 100%;"
              tabindex="-1"
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-81qg8w"
              >
                <span>
                  Response
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <br />
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-74d805-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Add
        </button>
      </div>
      <br />
      <p
        class="MuiTypography-root MuiTypography-body1 css-rizt0-MuiTypography-root"
      >
        Bot is 
        stopped
      </p>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSuccess MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorSuccess MuiButton-root MuiButton-contained MuiButton-containedSuccess MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorSuccess css-1elmdwi-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Start
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedError MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorError Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedError MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorError css-1f7qp0x-MuiButtonBase-root-MuiButton-root"
        disabled=""
        tabindex="-1"
        type="button"
      >
        Stop
      </button>
    </div>
    <footer
      class="footer MuiBox-root css-0"
    >
      <div
        class="MuiContainer-root MuiContainer-maxWidthSm css-zzzaw2-MuiContainer-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body2 text_box css-7ctihp-MuiTypography-root"
        >
          Copyright Â©
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways url css-i9kmoy-MuiTypography-root-MuiLink-root"
            href="https://jh123x.com/"
          >
            Jh123x
          </a>
          2025
        </p>
      </div>
    </footer>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": Object {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
